start-tests.sh

```diff
@@ -7,106 +7,267 @@
+
+if [[ ! -z $CYPRESS_OPTIONS_HUB_PASSWORD && "$CYPRESS_OPTIONS_HUB_PASSWORD" != "null" ]]; then
+  log_color "cyan" "Logging into Kube API server."
+
+  export KUBECONFIG=./kube/config/hub-kubeconfig
+  touch $KUBECONFIG
+
+  oc login --server=https://api.${CYPRESS_OPTIONS_HUB_BASEDOMAIN}:6443 -u $CYPRESS_OPTIONS_HUB_USER -p $CYPRESS_OPTIONS_HUB_PASSWORD --insecure-skip-tls-verify
+  export OPTIONS_HUB_KUBECONFIG=$KUBECONFIG
+
+  if [[ -f $OPTIONS_HUB_KUBECONFIG ]]; then
+    echo -e "Succesfully detected hub cluster kubeconfig.\n"
+  else
+    echo -e "Failed to create or locate hub cluster kubeconfig.\n"
+  fi
 
-if [[ -z $OPTIONS_MANAGED_BASEDOMAIN || -z $OPTIONS_MANAGED_USER || -z $OPTIONS_MANAGED_PASSWORD ]]; then
-   log_color "yellow" "One or more variables are undefined. Copying kubeconfigs...\n"
-   cp /opt/.kube/import-kubeconfig ./config/import-kubeconfig
-   echo $OPTIONS_KUBECONFIG_MOUNT_PATH
+  export CYPRESS_OPTIONS_HUB_KUBECONFIG=$OPTIONS_HUB_KUBECONFIG
+fi
+
+VERSION=`oc get subscriptions.operators.coreos.com -A -o yaml | grep currentCSV:\ advanced-cluster-management | awk '{$1=$1};1' | sed "s/currentCSV:\ advanced-cluster-management.v//"`
+log_color "green" "Testing with ACM Version": "$VERSION\n"
+
+log_color "cyan" "Checking RedisGraph deployment."
+installNamespace=`oc get subscriptions.operators.coreos.com --all-namespaces | grep advanced-cluster-management | awk '{print $1}'`
+rgstatus=`oc get srcho searchoperator -o jsonpath="{.status.deployredisgraph}" -n ${installNamespace}`
+
+if [ "$rgstatus" == "true" ]; then
+  echo -e "RedisGraph deployment is enabled.\n"
+else
+  echo -e "RedisGraph deployment disabled, enabling and waiting 60 seconds for the search-redisgraph-0 pod.\n"
+  oc set env deploy search-operator DEPLOY_REDISGRAPH="true" -n $installNamespace
+  sleep 60
+fi
+
+# Search for managed clusters.
+MANAGED_CLUSTERS=($(oc get managedclusters -o custom-columns='name:.metadata.name' --no-headers))
+
+# Check to see if there are any managed cluster available.
+if [[ ${#MANAGED_CLUSTERS[@]} == "1" && ${MANAGED_CLUSTERS[0]} =~ "local-cluster" ]]; then
+  echo -e "No managable clusters detected for the hub cluster: $CYPRESS_OPTIONS_HUB_BASEDOMAIN.\n"
+  export CYPRESS_SKIP_MANAGED_CLUSTER_TEST=true
 else
-  log_color "cyan" "Logging into the managed cluster using credentials and generating the kubeconfig..."
-  mkdir ./import-kubeconfig && touch ./import-kubeconfig/kubeconfig
-  export KUBECONFIG=$(pwd)/import-kubeconfig/kubeconfig
-  export OPTIONS_MANAGED_URL="https://api.$OPTIONS_MANAGED_BASEDOMAIN:6443"
-  oc login --server=$OPTIONS_MANAGED_URL -u $OPTIONS_MANAGED_USER -p $OPTIONS_MANAGED_PASSWORD --insecure-skip-tls-verify
-  unset KUBECONFIG
-  log_color "yellow" "Copying managed cluster kubeconfig to ./cypress/config/import-kubeconfig ...\n"
-  cp ./import-kubeconfig/* ./config/import-kubeconfig
+  echo -e "Detected clusters within the fleet: ${GREEN}${MANAGED_CLUSTERS[@]}${NC}\n"
```

Comment:
Checking to see if there are any `managed` clusters available. By default `local-cluster` will always be listed; so if there is only one cluster listed and that is `local-cluster` or if there is only one cluster returned in general, we'll skip the `managed` cluster test.